lib/faraday/adapter/net_http_persistent.rb

```diff
@@ -17,8 +17,11 @@ def with_net_http_connection(env)
             define_method(:password) { proxy[:password] }
           end if proxy[:user]
         end
-
-        yield Net::HTTP::Persistent.new 'Faraday', proxy_uri
+        if Net::HTTP::Persistent::VERSION >= '3'
```

Comment:
🙇🏻

I switched this to open the possibility of adding arbitrary configuration keyword argument to the constructor.  Keeping the few positional arguments seemed silly.